x-pack/plugins/endpoint/server/mocks.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const createMockIndexPatternRetriever = (indexPattern: string) => {
+  const mockGetFunc = jest.fn().mockResolvedValue(indexPattern);
+  return {
+    getIndexPattern: mockGetFunc,
+    getEventIndexPattern: mockGetFunc,
+    getMetadataIndexPattern: mockGetFunc,
+  };
+};
+
+export const MetadataIndexPattern = 'metrics-endpoint-*';
+
+export const createMockMetadataIPRetriever = () => {
```

Comment:
is `createMockMetadataIPRetriever` because it clashes with `createMockIndexPatternRetriever`. Perhaps good to find a way to differentiate them while leaving `IndexPattern` spelt out.

Reply:
Yeah I can do `createMockMetadataIndexPatternRetriever` it was just getting long haha.